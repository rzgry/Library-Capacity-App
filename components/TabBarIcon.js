import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'expo';

import Colors from '../constants/Colors';

const TabBarIcon = ({ name, focused }) => (
  <Icon.Ionicons
    name={name}
    size={26}
    style={{ marginBottom: -3 }}
    color={focused ? Colors.tabIconSelected : Colors.tabIconDefault}
  />
);

TabBarIcon.propTypes = {
  name: PropTypes.string.isRequired,
  focused: PropTypes.bool,
};

TabBarIcon.defaultProps = {
  focused: false,
};

export default TabBarIcon;
