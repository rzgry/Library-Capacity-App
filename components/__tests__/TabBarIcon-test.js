import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { shallow } from 'enzyme';
import { Icon } from 'expo';
import TabBarIcon from '../TabBarIcon';
import Colors from '../../constants/Colors';

describe('<TabBarIcon />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TabBarIcon name="md-information-circle" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('changes color when focused', () => {
    const wrapper = shallow(<TabBarIcon name="md-information-circle" />);
    expect(wrapper.find(Icon.Ionicons)).toHaveLength(1);

    expect(wrapper.find(Icon.Ionicons).props().color).toBe(Colors.tabIconDefault);
    wrapper.setProps({ focused: true });
    expect(wrapper.find(Icon.Ionicons).props().color).toBe(Colors.tabIconSelected);
  });
});
