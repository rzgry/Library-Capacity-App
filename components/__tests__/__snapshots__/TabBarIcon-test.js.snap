// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabBarIcon /> renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;
